### PR TITLE
Chapter 11

### PR DESCRIPTION
**Pull request by shewlyakov**

### CODE REVIEW

File: app/assets/stylesheets/account_activations.scss [hunk 1]
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the AccountActivations controller here.
```
Comment (bohdandalekorey) [inline app/assets/stylesheets/account_activations.scss]: Remove empty file!

File: app/controllers/account_activations_controller.rb [hunk 2]
```diff
@@ -0,0 +1,16 @@
+class AccountActivationsController < ApplicationController
+  def edit
+    user = User.find_by(email: params[:email])
+
+    if user && !user.activated? && user.authenticated?(:activation, params[:id])
+      user.activate
+      log_in user
+      flash[:success] = "Account a1ctivated!"
+      redirect_to user
+
```
Comment (bohdandalekorey) [inline app/controllers/account_activations_controller.rb]: unnecessary empty line

File: app/controllers/account_activations_controller.rb [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+    else
+      flash[:danger] = "Invalid activation link"
```
Comment (bohdandalekorey) [inline app/controllers/account_activations_controller.rb]: use single quotes

File: app/controllers/account_activations_controller.rb [hunk 4]
```diff
@@ -0,0 +1,16 @@
+class AccountActivationsController < ApplicationController
+  def edit
+    user = User.find_by(email: params[:email])
+
+    if user && !user.activated? && user.authenticated?(:activation, params[:id])
+      user.activate
+      log_in user
+      flash[:success] = "Account a1ctivated!"
```
Comment (bohdandalekorey) [inline app/controllers/account_activations_controller.rb]: use single quotes

File: app/controllers/sessions_controller.rb [hunk 5]
```diff
@@ -5,9 +5,17 @@ def create
...
-      log_in user
-      params[:session][:remember_me] == '1' ? remember(user) : forget(user)
-      redirect_back_or user
+      if user.activated?
+        log_in user
+        params[:session][:remember_me] == '1' ? remember(user) : forget(user)
+        redirect_back_or user
+      else
+        message = 'Account not activated.'
+        message += 'Check your email for the activation link.'
```
Comment (bohdandalekorey) [inline app/controllers/sessions_controller.rb]: Why this addition to the string? You can do this in one line

File: app/helpers/account_activations_helper.rb [hunk 6]
```diff
@@ -0,0 +1,2 @@
+module AccountActivationsHelper
```
Comment (bohdandalekorey) [inline app/helpers/account_activations_helper.rb]: Remove empty files

File: app/mailers/user_mailer.rb [hunk 7]
```diff
@@ -0,0 +1,12 @@
+class UserMailer < ApplicationMailer
+  def account_activation(user)
+    @user = user
+    mail to: user.email, subject: 'Account activation'
+  end
+
+  def password_reset
+    @greeting = 'Hi'
+
```
Comment (bohdandalekorey) [inline app/mailers/user_mailer.rb]: I think it's superfluous.

File: app/views/user_mailer/account_activation.html.erb [hunk 8]
```diff
@@ -0,0 +1,7 @@
+<h1>Sample App</h1>
+<p>Hi <%= @user.name %>,</p>
+<p>
+Welcome to the Sample App! Click on the link below to activate
```
Comment (bohdandalekorey) [inline app/views/user_mailer/account_activation.html.erb]: Incorrect tabulation

File: config/environments/development.rb [hunk 9]
```diff
@@ -1,4 +1,4 @@
-require "active_support/core_ext/integer/time"
+require 'active_support/core_ext/integer/time'
```
Comment (bohdandalekorey) [inline config/environments/development.rb]: Unnecessary changes

File: db/migrate/20210618183407_add_activation_to_user.rb [hunk 10]
```diff
@@ -0,0 +1,8 @@
+class AddActivationToUser < ActiveRecord::Migration[6.1]
+  def change
+    add_column :users, :s, :string
```
Comment (bohdandalekorey) [inline db/migrate/20210618183407_add_activation_to_user.rb]: What is the "s" column?

File: test/mailers/previews/user_mailer_preview.rb [hunk 11]
```diff
@@ -0,0 +1,16 @@
...
+    user = User.last
+    user.activation_token = User.new_token
+    UserMailer.account_activation(user)
+  end
+
+  # Preview this email at http://localhost:3000/rails/mailers/user_mailer/password_reset
+  def password_reset
+    UserMailer.password_reset
+  end
+
```
Comment (bohdandalekorey) [inline test/mailers/previews/user_mailer_preview.rb]: Remove this line

File: test/controllers/account_activations_controller_test.rb [hunk 12]
```diff
@@ -0,0 +1,7 @@
+require "test_helper"
+
+class AccountActivationsControllerTest < ActionDispatch::IntegrationTest
+  # test "the truth" do
```
Comment (bohdandalekorey) [inline test/controllers/account_activations_controller_test.rb]: What it is?

Comment (shewlyakov) [reply test/controllers/account_activations_controller_test.rb]: automatic generated file

**Review by bohdandalekorey — APPROVED** [review]
